### PR TITLE
Added Typescript definition to library

### PR DESCRIPTION
**Pull request by DenisCarriere**
Happy new year! 🎉

Included the Typescript definition to this repo.

@mourner 

FYI: Purposely didn't update `@turf/turf` package since it breaks the tests.

### CODE REVIEW

**Comment by mourner** [comment]
Since I'm not a TypeScript user and won't be maintaining the definition, could this stay at `DefinitelyTyped`? I know it's more convenient to have it here, but this really increases the maintenance burden for me.

**Comment by DenisCarriere** [comment]
@mourner It's up to you, I personally will make sure to maintain this. It's not very often that the `cheap-ruler` functionality change or break backwards compatibility.

How about a compromise... I strip off all the comments and leave only the `index.d.ts` core methods?

**Comment by DenisCarriere** [comment]
@mourner I've converted the definition file to be "low maintenance", since all your documentation & JSDocs is already inside your `.js` source code.

It's 35 lines of code now, very easy to manage.

```typescript
declare function cheapRuler(lat: number, units?: string): cheapRuler.CheapRuler;
declare namespace cheapRuler {
    type BBox = [number, number, number, number]
    type Point = [number, number]
    type Line = Point[]
    type Points = Point[]
    type Polygon = Point[][]

    class CheapRuler {
        distance(a: cheapRuler.Point, b: Point): number;
        bearing(a: Point, b: Point): number;
        destination(p: Point, dist: number, bearing: number): Point;
        lineDistance(points: Points): number;
        area(polygon: Polygon): number;
        along(line: Line, dist: number): Point;
        pointOnLine(line: Line, p: Point): {point: Point, index: number, t: number};
        lineSlice(start: Point, stop: Point, line: Line): Line;
        lineSliceAlong(start: number, stop: number, line: Line): Line;
        bufferPoint(p: Point, buffer: number): BBox;
        bufferBBox(bbox: BBox, buffer: number): BBox;
        insideBBox(p: Point, bbox: BBox): boolean;
    }
    const units: {
        kilometers: number;
        miles: number;
        nauticalmiles: number;
        meters: number;
        metres: number;
        yards: number;
        feet: number;
        inches: number;
    }
    function fromTile(y: number, z: number, units?: string): CheapRuler;
}
export = cheapRuler;
```

File: test/types.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+ruler.destination(point1, 0.1, 90)
+ruler.area(polygon)
+ruler.along(line, 2.5)
+ruler.lineDistance(line)
+ruler.pointOnLine(line, point1).point
+ruler.lineSlice(point1, point2, line)
+ruler.lineSliceAlong(10, 20, line)
+ruler.bufferPoint(point1, 0.01)
+ruler.bufferBBox([30.5, 50.5, 31, 51], 0.2)
+ruler.insideBBox([30.5, 50.5], [30, 50, 31, 51])
```
Comment (mourner) [inline test/types.ts]: How does this test file get run?

File: package.json [hunk 2]
```diff
@@ -12,7 +13,8 @@
     "istanbul": "^0.4.3",
     "node-vincenty": "0.0.6",
     "tape": "^4.5.1",
-    "turf": "^2.0.2"
+    "turf": "^2.0.2",
+    "typescript": "^2.1.4"
```
Comment (mourner) [inline package.json]: How is the typescript dev dependency used?

File: test/types.ts (hunk 1 above)
Comment (DenisCarriere) [inline test/types.ts]: DefinitelyType is doing this by simply running `tsc types.ts` if no errors show up it's a pass, some way a linter would be run.

Comment (DenisCarriere) [inline test/types.ts]: More info on their Contributing Info page.
http://definitelytyped.org/guides/contributing.html#tests

File: package.json (hunk 2 above)
Comment (DenisCarriere) [inline package.json]: Would be used for testing, I can include it in your `package.json`
```json
  "scripts": {
    "pretest": "eslint index.js bench test/*.js && npm run types",
    "test": "tape test/test.js",
    "types": "tsc test/types.ts",
    "build": "browserify index.js -s cheapRuler > cheap-ruler.js",
    "prepublish": "npm run build",
    "cov": "istanbul cover tape test/test.js"
  },
```

**Comment by DenisCarriere** [comment]
@mourner Added Typescript tests to your `package.json`.

Commit https://github.com/mapbox/cheap-ruler/pull/23/commits/3e557b1418640fe39fd9b65b9f2c8bb792b920ee

For example, if I would change the following

```typescript
type Point = [number, number]
```
```typescript
type Point = string
```

Run the tests, it would cause an error like this.

```
$ tsc test/types.ts
test/types.ts(7,7): error TS2322: Type 'number[]' is not assignable to type 'string'
.
test/types.ts(8,7): error TS2322: Type 'number[]' is not assignable to type 'string'
.
test/types.ts(9,7): error TS2322: Type 'number[][][]' is not assignable to type 'str
ing[][]'.
  Type 'number[][]' is not assignable to type 'string[]'.
    Type 'number[]' is not assignable to type 'string'.
test/types.ts(13,7): error TS2322: Type 'number[][]' is not assignable to type 'stri
ng[]'.
  Type 'number[]' is not assignable to type 'string'.
test/types.ts(34,18): error TS2345: Argument of type 'number[]' is not assignable to
 parameter of type 'string'.
```

**Comment by DenisCarriere** [comment]
One of the main reasons why I'm including this file inside your repo is because I'm tired of importing tons of `@types` module, there's already enough imported JS packages in todays age. My project's `devDependencies` is growing every week.

Also, **maybe** one day, Flow & Typescript will be compatible with each to other.

**Comment by mourner** [comment]
Thanks! The PR look great now.

**Pull request merged**